src/pay/payment_producer.py

```diff
@@ -250,7 +250,7 @@ def run(self):
 
         return
 
-    def try_to_pay(self, pymnt_cycle, expected_rewards=False):
```

Comment:
For the python noob. The variable `expected_rewards` was only defined as part of the parameter in the `try_to_pay` method and you removed it by referencing to the `self.reward_type` ?

Kind of horrible that an additional boolean has been introduced instead of using the reward types from the constants. I am glad this has been removed

Reply:
Yes, you are right.